### PR TITLE
lib/strings/toShellVars: handle derivations as strings

### PR DESCRIPTION
**Pull request by ncfavier**

### CODE REVIEW

File: lib/strings.nix [hunk 1]
```diff
@@ -356,7 +356,7 @@ rec {
   */
   toShellVar = name: value:
     lib.throwIfNot (isValidPosixName name) "toShellVar: ${name} is not a valid shell variable name" (
-    if isAttrs value then
+    if isAttrs value && ! isCoercibleToString value then
```
Comment (roberth) [inline lib/strings.nix:359]: Needs doc update, like

```diff
-     Supported value types are strings (mapped to regular variables), lists of strings
+     Supported value types are strings (mapped to regular variables), string-coercible attrsets
+     (also mapped to regular variables), lists of strings
```

File: lib/tests/misc.nix [hunk 2]
```diff
@@ -269,6 +269,7 @@ runTests {
           strings
           possibly newlines
         '';
+        drv.outPath = "/path";
```
Comment (roberth) [inline lib/tests/misc.nix:272]: What happens to a path expr?
Does `__toString` work?

```suggestion
        drv.outPath = "/path";
        drv.foo = "ignore drv attr";
        pathValue = ./misc.nix;
        stringable.__toString = _: "hello toString";
        stringable.foo = "ignore stringable attr";
```

**Review by roberth — COMMENTED** [review]
It's a good change. Glad that you catch this while the function is young; saves some concerns.
Just need to fill in some details and we're good.

File: lib/strings.nix (hunk 1 above)
Comment (roberth) [reply lib/strings.nix:360]: Did you change it? The current text is too implicit.

It's also better to use singular where possible, as it avoids ambiguity.

Comment (ncfavier) [reply lib/strings.nix:360]: I think it's just the right amount of implicit: the intent is that strings are mapped to regular variables, and, as a side-note, things that should obviously be treated as strings are treated as strings. Mentioning string-coercible attrsets feels heavy and impedes readability IMO.

:+1: for singular

Comment (roberth) [reply lib/strings.nix:360]: Fair enough.

**Review by roberth — APPROVED** [review]

**Pull request merged**